xdsl/dialects/llvm.py

```diff
@@ -610,6 +610,106 @@ def verify(self, verify_nested_ops: bool = True):
         super().verify(verify_nested_ops)
 
 
+class ICmpPredicateFlag(StrEnum):
+    EQ = "eq"
+    NE = "ne"
+    SLT = "slt"
+    SLE = "sle"
+    SGT = "sgt"
+    SGE = "sge"
+    ULT = "ult"
+    ULE = "ule"
+    UGT = "ugt"
+    UGE = "uge"
+
+    @classmethod
+    def from_int(cls, i: int) -> ICmpPredicateFlag:
+        if i == 0:
+            return ICmpPredicateFlag.EQ
+        elif i == 1:
+            return ICmpPredicateFlag.NE
+        elif i == 2:
+            return ICmpPredicateFlag.SLT
+        elif i == 3:
+            return ICmpPredicateFlag.SLE
+        elif i == 4:
+            return ICmpPredicateFlag.SGE
+        elif i == 5:
+            return ICmpPredicateFlag.SGT
+        elif i == 6:
+            return ICmpPredicateFlag.ULT
+        elif i == 7:
+            return ICmpPredicateFlag.ULE
+        elif i == 8:
+            return ICmpPredicateFlag.UGE
+        elif i == 9:
+            return ICmpPredicateFlag.UGT
```

Comment:
There has to be a better way of doing this, I'd expect some helper to exist in the Python enum implementation, otherwise we should cache this mapping.